### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by c-git**
- Cleaned up items marked as pub. Removed pub if not needed and put `pub (crate)` if only available inside of the crate.
- Created documentation for all public items in the crate
- Added lint to ensure all items exposed for solution development have a debug impl
- Added a rustfmt.toml to document how to get the wrapping to work
- Resolves (https://github.com/rust-practice/cargo-leet/discussions/20)

### CODE REVIEW

File: src/leetcode_env/list.rs [hunk 1]
```diff
@@ -79,6 +72,7 @@ impl From<Vec<i32>> for ListHead {
     }
 }
 
+// TODO: Test the happy path of going from a linked list to a vec
```
Comment (sak96) [inline src/leetcode_env/list.rs:75]: TODO ?

Comment (c-git) [reply src/leetcode_env/list.rs:75]: Yes it's a note for all the places I think need tests.

File: src/lib.rs [hunk 2]
```diff
@@ -1,6 +1,49 @@
...
+//! the associated help is probably the best way to get an idea of how to use
+//! the tool. Screenshots of the help can be found in the
+//! [readme](https://github.com/rust-practice/cargo-leet#screenshots) on github.
+//! For the sake of maintainability features added will be documented there
+//! instead of always needing to update multiple places.
+//!
+//! ### Leetcode Environment Support
+//!
+//! **cargo-leet** also includes helper code with structs and traits to simulate
+//! the environment that you code would run in on the leetcode servers so that
```
Comment (nassersaazi) [inline src/lib.rs:23]:
```suggestion
//! the environment that your code would run in on the leetcode servers so that
```

File: src/lib.rs [hunk 3]
```diff
@@ -1,6 +1,49 @@
...
+//! you are able to run tests on your code locally. It also provides a few extra
+//! types that facilitate testing especially as it relates to creating test
+//! cases from the text provided by leetcode.
+//!
+//! ## Feature flags
+//!
+//! **cargo-leet** uses feature flags to control which code gets compiled based
+//! on how the create is being used. This is especially important for the code
```
Comment (nassersaazi) [inline src/lib.rs:31]:
```suggestion
//! on how the crate is being used. This is especially important for the code
```

File: src/lib.rs [hunk 4]
```diff
@@ -1,6 +1,49 @@
...
+//! imported in the solution repository as this repo may be using a much older
+//! version of the rust toolchain due to the fact that leetcode uses a much
+//! older version on their servers and some users may want to use the same
+//! version to ensure their code will always work upon upload. However, because
+//! it is such an old version many of the creates used in the development of the
```
Comment (nassersaazi) [inline src/lib.rs:36]:
```suggestion
//! it is such an old version many of the crates used in the development of the
```

**Review by nassersaazi — APPROVED** [review]

**Pull request merged**